test/utils/AATest.sol

```diff
@@ -32,6 +33,17 @@ abstract contract AATest is Test {
             });
     }
 
+    function handleUserOp(UserOperation memory userOp) internal {
+        UserOperation[] memory ops = new UserOperation[](1);
+        ops[0] = userOp;
+
+        entryPoint.handleOps(ops, payable(msg.sender));
+    }
+
+    function expectRevertFailedOp(string memory reason) internal {
+        vm.expectRevert(abi.encodeWithSelector(IEntryPoint.FailedOp.selector, 0, reason));
```

Comment:
What does the `0` mean?

Reply:
First arg for error `FailedOp` is `opIndex`, an indicator for which user op is failed. 

In tests most of the time we would only send one user op and `0` is  enough to cover our current needs. We can overload `expectRevertFailedOp` when we indeed have to check for multiple user ops.